src/test/java/com/facebook/openwifirrm/modules/ModelerUtilsTest.java

```diff
@@ -547,4 +551,289 @@ void testPostDot11nAggregatedWifiScanEntry() {
+			1,
+			10,
+			100,
+			bssidA,
+			stationA1,
+			new int[] { 10, 20, 30 }
+		);
+		AggregatedState aggStateA2 = TestUtils.createAggregatedState(
+			2,
+			10,
+			100,
+			bssidA,
+			stationA2,
+			new int[] { 20, 30, 40 }
+		);
+		AggregatedState aggStateB = TestUtils.createAggregatedState(
+			3,
+			20,
+			200,
+			bssidB,
+			stationB,
+			new int[] { 10, 20, 30 }
+		);
+		AggregatedState aggStateC = TestUtils.createAggregatedState(
+			1,
+			10,
+			100,
+			bssidC,
+			stationC,
+			new int[] { 100, 200, 300 }
+		);
+
+		Map<String, List<AggregatedState>> bssidToAggregatedStates =
+			new HashMap<>();
+		bssidToAggregatedStates.put(
+			ModelerUtils.getBssidStationKeyPair(bssidA, stationA1),
+			new ArrayList<>(Arrays.asList(aggStateA1))
+		);
+
+		bssidToAggregatedStates.put(
+			ModelerUtils.getBssidStationKeyPair(bssidA, stationA2),
+			new ArrayList<>(Arrays.asList(aggStateA2))
+		);
+		bssidToAggregatedStates.put(
+			ModelerUtils.getBssidStationKeyPair(bssidB, stationB),
+			new ArrayList<>(Arrays.asList(aggStateB))
+		);
+		bssidToAggregatedStates.put(
+			ModelerUtils.getBssidStationKeyPair(bssidC, stationC),
+			new ArrayList<>(Arrays.asList(aggStateC))
+		);
+
+		State toBeAggregated1 = TestUtils.createState(
+			2,
+			10,
+			100,
+			bssidA,
+			new String[] { stationA1, stationA1, stationA2 },
+			new int[] { 40, 50, 60 },
+			1
+		);
+
+		ModelerUtils
+			.addStateToAggregation(bssidToAggregatedStates, toBeAggregated1);
+
+		assertEquals(
+			bssidToAggregatedStates
+				.get(ModelerUtils.getBssidStationKeyPair(bssidA, stationA1))
+				.get(0).rssi,
+			Arrays.asList(10, 20, 30)
+		);
+
+		assertEquals(
+			bssidToAggregatedStates
+				.get(ModelerUtils.getBssidStationKeyPair(bssidA, stationA1))
+				.get(1).rssi,
+			Arrays.asList(40, 50)
+		);
+		assertEquals(
+			bssidToAggregatedStates
+				.get(ModelerUtils.getBssidStationKeyPair(bssidA, stationA2))
+				.get(0).rssi,
+			Arrays.asList(20, 30, 40, 60)
+		);
+
+		State toBeAggregated2 = TestUtils.createState(
+			3,
+			20,
+			200,
+			bssidB,
+			new String[] { stationB },
+			new int[] { 40 },
+			1
+		);
+		ModelerUtils
+			.addStateToAggregation(bssidToAggregatedStates, toBeAggregated2);
+		assertEquals(
+			bssidToAggregatedStates
+				.get(ModelerUtils.getBssidStationKeyPair(bssidB, stationB))
+				.get(0).rssi,
+			Arrays.asList(10, 20, 30, 40)
+		);
+	}
+
+	@Test
+	void testGetAggregatedStates() {
+		final long obsoletionPeriodMs = 60000000;
+		final String serialNumberA = "aaaaaaaaaaaa";
+		final String bssidA = "aa:aa:aa:aa:aa:a";
+		final String serialNumberB = "bbbbbbbbbbbb";
+		final String bssidB = "bb:bb:bb:bb:bb:bb";
+		final String serialNumberC = "cccccccccccc";
+		final String bssidC = "cc:cc:cc:cc:cc:cc";
+
+		long refTimeMs = TestUtils.DEFAULT_LOCAL_TIME;
+
+		DataModel dataModel = new DataModel();
+
+		// This serie of StateA is used to test a valid input states.
+		State time1ToStateA = TestUtils.createState(
```

Comment:
I think just `time1StateA` is better (i.e., remove the `To`)? What do you think? For this and all the other similar vars.